similar/main/game.cpp

```diff
@@ -1307,12 +1307,15 @@ window_event_result game_handler(window *,const d_event &event, const unused_win
 			Game_wind = NULL;
 			event_toggle_focus(0);
 			key_toggle_repeat(1);
+			return window_event_result::ignored;
 			break;
 
 		default:
 			break;
 	}
-	return window_event_result::ignored;
+	
+	// If we deleted the window, tell the event loop
+	return Game_wind ? window_event_result::ignored : window_event_result::deleted;
```

Comment:
That seems like a reasonable choice to reduce churn, but I am still confused as to what this return achieves.  Is it expected that it takes the `deleted` branch because the window already handled a `EVENT_WINDOW_CLOSE` message, set `Game_wind` to null, returned, and now another message has been delivered to that window?  If not, under what circumstances are you expecting to reach that return statement with `Game_wind` set to null?  My original comment was because, when `game_handler` is processing `EVENT_WINDOW_CLOSE`, you added a return in that case, so it will not reach the general purpose return at the bottom.